src/models/payments.js

```diff
@@ -251,13 +245,21 @@ function isTraderFunds (config, funds) {
 }
 
 function * updateSourceTransfer (updatedTransfer, traderCredit, ledgers, config) {
-  const destinationTransfer = traderCredit.memo && traderCredit.memo.destination_transfer
-  if (!destinationTransfer) return
-  ledgers.validateTransfer(destinationTransfer)
-
-  const isTransferReady = updatedTransfer.state === 'prepared' || updatedTransfer.state === 'executed'
+  const isTransferReady = updatedTransfer.state === 'prepared' ||
+    (updatedTransfer.state === 'executed' && !updatedTransfer.execution_condition)
   if (!isTransferReady) return
 
+  if (!config.ledgerCredentials[updatedTransfer.ledger]) {
+    throw new AssetsNotTradedError('This connector does not support the given asset pair')
+  }
+
+  // TODO this is cheating, but how do we know the type of the final transfer's ledger?
```

Comment:
Yep, the connector needs to know the `ledger` of the final transfer so that it knows how to route it.